### PR TITLE
Add ability to prepend query comments

### LINKED ISSUE #1982: Query comment
Hello,

Is there a way to add _sql comment_ to select query without using _raw queries_? I tried to look for docs on http://knexjs.org/ but can see comment support only for table manipulation. 

Query comments are useful for routing hints on maxscale router (https://mariadb.com/kb/en/mariadb-enterprise/mariadb-maxscale-14/maxscale-hint-syntax/).

### PR DESCRIPTION
**Pull request by francois2metz**
This is a rebased and updated version of #2815 that fixes #1982.

About parameter binding, I did tried to escape them, but I doesn't works for mysql. So I forbidden any ?.

### CODE REVIEW

**Comment by francois2metz** [comment]
I fixed a bug with cockroachdb as ids are not numeric. It should be fine now. I'll squash commit later.

**Comment by kibertoad** [comment]
@francois2metz Can you please send documentation PR to https://github.com/knex/documentation?

**Comment by francois2metz** [comment]
@kibertoad Hi! I created the documentation PR: https://github.com/knex/documentation/pull/450

File: test/integration2/query/select/selects.spec.js [hunk 1]
```diff
@@ -863,6 +863,17 @@ describe('Selects', function () {
         });
       });
 
+      it('#1982 - Allow comments in db', function() {
+        return knex('accounts')
```
Comment (OlivierCavadenti) [inline test/integration2/query/select/selects.spec.js:867]: do async / await tests

File: test/integration2/query/select/selects.spec.js [hunk 2]
```diff
@@ -863,6 +863,17 @@ describe('Selects', function () {
...
+          .comment('Integration Comment')
+          .select('first_name', 'email')
+          .limit(1)
+          .then(function(results) {
+            expect(results).to.have.length(1);
```
Comment (OlivierCavadenti) [inline test/integration2/query/select/selects.spec.js:872]: no way to test the content of the result ?

**Review by OlivierCavadenti — COMMENTED** [review]
Seems ok for me ! Thanks and sorry for the delay

File: test/integration2/query/select/selects.spec.js (hunk 2 above)
Comment (francois2metz) [reply test/integration2/query/select/selects.spec.js:872]: Not sure how I could do that.

**Review by JMHenri — APPROVED** [review]
Looks good to me!

**Review by OlivierCavadenti — APPROVED** [review]

**Comment by lindakung** [comment]
Very excited for this! Any ETA on when a new version will be released?

**Pull request merged**